src/test/java/io/strimzi/kafka/bridge/http/ProducerIT.java

```diff
@@ -1117,4 +1122,95 @@ void sendSimpleMessageWithWrongContentType(VertxTestContext context) throws Inte
+
+        JsonObject record = createDynamicRecord(key, value, firstContentType);
+        JsonArray records = new JsonArray().add(record);
+        JsonObject root = new JsonObject().put("records", records);
+
+        LOGGER.info("Sending:\n{}", root.toString());
+
+        // Send the first request
+        producerService()
+            .sendRecordsRequest(topic, root, firstContentType)
+            .sendJsonObject(root, response -> {
+                if (response.succeeded()) {
+                    HttpResponse<JsonObject> httpResponse = response.result();
+                    context.verify(() -> {
+                        assertThat(httpResponse.statusCode(), is(HttpResponseStatus.OK.code()));
+                        LOGGER.info("Successfully processed record with content type: {}", secondContentType);
+                    });
+                } else {
+                    context.failNow(response.cause());
+                }
+            });
+
+        record = createDynamicRecord(key, value, secondContentType);
+        records = new JsonArray().add(record);
+        root = new JsonObject().put("records", records);
+
+        // Send the second request
+        producerService()
+            .sendRecordsRequest(topic, root, secondContentType)
+            .sendJsonObject(root, response -> {
+                if (response.succeeded()) {
+                    HttpResponse<JsonObject> httpResponse = response.result();
+                    context.verify(() -> {
+                        assertThat(httpResponse.statusCode(), is(HttpResponseStatus.OK.code()));
+                        LOGGER.info("Successfully processed record with content type: {}", secondContentType);
+                        context.completeNow();
+                    });
+                } else {
+                    context.failNow(response.cause());
+                }
+            });
+
+        assertThat(context.awaitCompletion(TEST_TIMEOUT, TimeUnit.SECONDS), is(true));
+    }
+
+    private JsonObject createDynamicRecord(JsonObject key, JsonObject value, String contentType) {
+        JsonObject record = new JsonObject();
+
+        switch (contentType) {
+            case BridgeContentType.KAFKA_JSON_JSON:
+                // Key and value are already JSON objects
+                record.put("key", key);
+                record.put("value", value);
+                break;
+            case BridgeContentType.KAFKA_JSON_BINARY:
+                record.put("key", Base64.getEncoder().encodeToString(key.encode().getBytes()));
+                record.put("value", Base64.getEncoder().encodeToString(value.encode().getBytes()));
+                break;
+            case BridgeContentType.KAFKA_JSON_TEXT:
+                // Convert JSON objects to string representations
+                record.put("key", key.encode());
+                record.put("value", value.encode());
```

Comment:
Have you verified what this `encode` return when the `JsonObject` is just a string? I mean ... a JSON string includes quotes so for example the String "foo" will be `"foo"`, while a plain text will be just `foo`. I am worried that the `encode` is returning a JSON string which is not what we want, but a plain one.